.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,103 @@
+name: Test & Publish
+
+# Explicit permissions for security
+permissions:
+  contents: read
+  pull-requests: write
+  id-token: write
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
we should remove pull_request, right? we only want this running on main or in emergency situations on workflow_dispatch